### PR TITLE
@predictions directive docs

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cli-toolchain/graphql.md [hunk 1]
```diff
@@ -2379,6 +2379,69 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+The supported actions in this directive are included in the definition.
+
+```
+# where the parent this field is defined on is a query type
+        directive @predictions(actions: [PredictionsActions!]!) on FIELD_DEFINITION
+        enum PredictionsActions {
+          identifyText
+          identifyLabels
+          convertTextToSpeech
+          translateText
```
Comment (dabit3) [inline cli-toolchain/graphql.md]: Looks like it's missing a closing `}`

Comment (SwaySway) [reply cli-toolchain/graphql.md]: added closing bracket

**Comment by SwaySway** [comment]
> Talked to Josue and provided overall feedback:
> 
>     1. Provide more context into what the @predictions directive is actually doing in the Usage section
> 
>     2. Provided frontend code needs to be stated more explicitly.
> 
> 
> Will do another language edit when the new version is submitted.

Added the following
- Backend docs explanation
- JS snippet
- Section detailing what actions are available to use

File: cli-toolchain/graphql.md [hunk 2]
```diff
@@ -2379,6 +2379,157 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+      targetLanguage: "es"
+    }
+    convertTextToSpeech: {
+      voiceID: "Conchita"
+    }
+  })
+}
+```
+
+A code example of this using the JS Library:
```
Comment (dabit3) [inline cli-toolchain/graphql.md]: I noticed in the regular Predictions category docs you have to [configure a pluggable](https://aws-amplify.github.io/docs/js/predictions#configure-your-app), is this not necessary when using this new API?

File: cli-toolchain/graphql.md [hunk 3]
```diff
@@ -2379,6 +2379,157 @@ Here is a complete list of searchable operations per GraphQL type supported as o
 | Boolean | `eq`, `ne`      |
 
 
+
+### @predictions
+
+The `@predictions` directive allows one to query an orchestration of AI/ML services such as Amazon Rekognition, Amazon Translate, and/or Amazon Polly.
```
Comment (dabit3) [inline cli-toolchain/graphql.md]: Would it make sense to link back to [the docs of how to add predictions](https://aws-amplify.github.io/docs/js/predictions#predictions) in some way?

File: cli-toolchain/graphql.md (hunk 2 above)
Comment (SwaySway) [reply cli-toolchain/graphql.md]: Nope this is not necessary - the resources created are used so AppSync has permissions to make those calls on behalf of the caller.

File: cli-toolchain/graphql.md (hunk 3 above)
Comment (SwaySway) [reply cli-toolchain/graphql.md]: I suppose we could to outline the services being used though implementation is different.

File: cli-toolchain/graphql.md [hunk 4]
```diff
@@ -2379,6 +2379,157 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+        < SpeakLabels />
+    </div>
+  );
+}
+export default App;
+```
+
+#### What happens in the backend
+
+From the provided schema above the action list `[ identifyText translateText convertTextToSpeech ]`. The following resources are created to communicate with Amazon Rekognition, Translate and Polly.
```
Comment (swaminator) [inline cli-toolchain/graphql.md]: The example schema will communicate with Amzn Reko.... (actions: `[ identifyText translateText convertTextToSpeech ]`) to generate the following resources: 
- IAM Policy for each service (e.g. Amazon Rekognition `detectText` policy)
- An AppSync VTL function for each action
- A data source for each action

File: cli-toolchain/graphql.md [hunk 5]
```diff
@@ -2379,6 +2379,157 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+
+- IAM Policy for each service
+  - e.g. Amazon Rekognition `detectText` Policy
+- An AppSync Function for each action
+- A datasource for each action
+
+Finally a resolver is created for `speakTranslatedImageText` which is a pipeline resolver composed of AppSync functions which are defined by the action list provided in the directive.
+
+#### Allowed Actions
```
Comment (swaminator) [inline cli-toolchain/graphql.md]: Actions

File: cli-toolchain/graphql.md [hunk 6]
```diff
@@ -2379,6 +2379,160 @@ Here is a complete list of searchable operations per GraphQL type supported as o
...
+For each action the following is created: 
+
+- IAM Policy for each service (e.g. Amazon Rekognition `detectText` Policy)
+- An AppSync VTL function
+- An AppSync DataSource
+
+Finally a resolver is created for `speakTranslatedImageText` which is a pipeline resolver composed of AppSync functions which are defined by the action list provided in the directive.
+
+#### Actions
+Valid Actions allowed in predictions are as follows:
```
Comment (nikhil-dabhade) [inline cli-toolchain/graphql.md]: We should also add that these 4  actions can be called individually as well. Right now we mention just convertTextToSpeech alone

**Pull request merged**